datafusion/common/src/scalar.rs

```diff
@@ -2575,7 +2575,7 @@ mod tests {
         // The alignment requirements differ across architectures and
         // thus the size of the enum appears to as as well
 
-        let expected = match cfg!(target_arch = "aarch64") {
+        let expected = match cfg!(target_arch = "x86_64") {
             true => 64,
             false => 48,
```

Comment:
Shouldn't they both be `64`?

Reply:
Actually, I am curious about how are `64` and `48` calculated here.

Reply:
I find a doc about the layout of `Enum` (https://rust-lang.github.io/unsafe-code-guidelines/layout/enums.html#enum-layout-rules), and will learn it carefully

Reply:
These 4 values seem to occupy largest memory
```rust
    TimestampSecond(Option<i64>, Option<String>), // 8 + 16 + 24
    /// Timestamp Milliseconds
    TimestampMillisecond(Option<i64>, Option<String>),
    /// Timestamp Microseconds
    TimestampMicrosecond(Option<i64>, Option<String>),
    /// Timestamp Nanoseconds
    TimestampNanosecond(Option<i64>, Option<String>),
```
The payload occupies 48 bytes and the tag occupies 1 byte (but this could be represented by niche?). So the overall size of the Enum if 48 I guess.

Reply:
> Shouldn't they both be `64`?

Maybe the both should be 48?